closed/src/java.base/share/classes/openj9/internal/security/RestrictedSecurity.java

```diff
@@ -758,15 +759,7 @@ private void initProviders() {
                 }
                 providerName = providerName.trim();
 
-                // Remove argument, e.g. -NSS-FIPS, if present.
-                pos = providerName.indexOf('-');
-                if (pos >= 0) {
-                    providerName = providerName.substring(0, pos);
-                }
```

Comment:
Why don't we need to keep this?

Reply:
Revert them, done.

Reply:
What about providers that must specify an argument, like in
```
RestrictedSecurity.NSS.140-2.securerandom.provider = SunPKCS11-NSS-FIPS
```
The reference to `SunPKCS11` should probably use the fully-qualified name, but `-NSS-FIPS` still needs to be removed to derive the class name.

Reply:
I don't think we can set as below:

`RestrictedSecurity.NSS.140-2.securerandom.provider = SunPKCS11-NSS-FIPS`

The provider name SunPKCS11-NSS-FIPS comes from two parts, provider name defined in its constructor + name defined in its configure file, the "${java.home}/conf/security/nss.fips.cfg" file. The SunPKCS11 constructor codes below:

```
    // Used by Secmod
    SunPKCS11(Config c) {
        super("SunPKCS11-" + c.getName(), PROVIDER_VER, c.getDescription());
        this.config = c;
```

When this provider is initialized with the configure file, for example the "nss.fips.cfg" file, the name will be SunPKCS11-NSS-FIPS. Without configure file, the name will be SunPKCS11.

The argument "-NSS-FIPS" is not a part of the name in its constructor. So I don't think we can specify the argument (without using the configure file) in the java.security provider list.

Reply:
My point is that the entry added to `providersFullyQualifiedClassName` is expected to be a fully-qualified class name, which would be `sun.security.pkcs11.SunPKCS11`, not `sun.security.pkcs11.SunPKCS11-NSS-FIPS`.

Reply:
Yes, that is correct. For the SunPKCS11 in NSS FIPS140-2 profile, it would be:

`RestrictedSecurity.NSS.140-2.jce.provider.1 = sun.security.pkcs11.SunPKCS11 ${java.home}/conf/security/nss.fips.cfg`

Reply:
Perhaps you misunderstand my concern. An existing properties file it might say:
```
RestrictedSecurity.NSS.140-2.jce.provider.1 = SunPKCS11-NSS-FIPS ...
```
I expected that this change will require that the class name will be fully-qualified, but the argument would still be present:
```
RestrictedSecurity.NSS.140-2.jce.provider.1 = sun.security.pkcs11.SunPKCS11-NSS-FIPS ...
```
Are you saying that must the argument be omitted?
```
RestrictedSecurity.NSS.140-2.jce.provider.1 = sun.security.pkcs11.SunPKCS11 ...
```
If so, it would seem there's no way to disallow other configurations of a given provider.

It's also not clear to me how the existing `java.properties` file could work properly for `SunPKCS11` because it does not specify the argument and so `RestrictedSecurity.isProviderAllowed("SunPKCS11-NSS-FIPS")` would return `false`.